File: javascript/armstrong-numbers/armstrong-numbers.js
export const isArmstrongNumber = (number) => {
    let digits = Math.trunc(Math.log10(number)) + 1;

    let n = number;
    let sum = 0;

    while (n > 0) {
        let currentDigit = n % 10;
        sum += Math.pow(currentDigit, digits);
        n = Math.trunc(n / 10);
    }

    return sum === number;
};

File: python/armstrong-numbers/armstrong_numbers.py
import math

def is_armstrong_number(number):
    return number == 0 or sum_of_digits(number) == number

def sum_of_digits(number):
    number_of_digits = int(math.log10(number) + 1)
    sum = 0

    while (number > 0):
        digit = number % 10
        sum += digit ** number_of_digits
        number //= 10

    return sum

